spec/main-process/appwindow-spec.js

```diff
@@ -0,0 +1,53 @@
+'use strict';
+
+var assert = require('assert');
+var sinon = require('sinon');
+var SandboxedModule = require('sandboxed-module');
+
+describe('AppWindow', function () {
+  var sandbox;
+  var AppWindow;
+  var fakeDialog;
+  var fakeBrowserWindow;
+  var fakeShell;
+  var appWindow;
+
+  beforeEach(function () {
+    sandbox = sinon.sandbox.create();
+    fakeBrowserWindow = sandbox.stub().returns({
+      loadUrl: sandbox.stub(),
+      show: sandbox.stub()
```

Comment:
The `on` method on the `fakeBrowserWindow` is missing. You may have to add more stubs, depending on test coverage.